src/main/java/net/neoforged/neoforge/client/entity/animation/AnimationTarget.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation;
+
+import net.minecraft.client.animation.AnimationChannel;
+import net.minecraft.client.animation.KeyframeAnimations;
+import org.joml.Vector3f;
+
+/**
+ * Wrapper for a {@link AnimationChannel.Target} and a way to transform a simple keyframe vector into a vector that
+ * makes sense for the given target.
+ * 
+ * @param channelTarget         The associated {@link AnimationChannel.Target}.
+ * @param keyframeTarget        An {@link AnimationKeyframeTarget} that transforms simple vectors into ones that make sense
+ *                              for the {@link #channelTarget}.
+ * @param inverseKeyframeTarget The inverse function of {@link #keyframeTarget}, used for serialization.
```

Comment:
Is there a reason for the serialization to be in the same format, or is it just for understandability?

Reply:
If the serialization didn't have transformations applied in inverse, that would have to be indicated somehow, and the output would also be very strange, as internal representations (like the fact that Y is inverted or that rotation is internally stored in radians) would leak.

Reply:
Hmm. This makes me wonder whether it makes more sense to keep the decoding method as is, but have a custom data provider for encoding that hold an object similar to the `AnimationDefinition` so such a transformation is unnecessary. I don't think the solution is wrong, I can just imagine some odd rounding issues.

Reply:
The rounding errors would be extremely minimal. Minecraft visual stuff often also uses numbers that are perfectly representable in base two, resulting in no rounding errors.